docs/build-with-0g/serving-network/provider.md

```diff
@@ -0,0 +1,130 @@
+---
+id: provider
+title: Provider
+sidebar_position: 2
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+To integrate a service into the 0G Serving Network, you need to transform your service into a **verifiable service** and connect it through the **provider broker container**.
+
+## Verifiable Services
+
+### Service Interface Requirements
+
+Your service need to adhere to the [OpenAI API Interface Standards](https://platform.openai.com/docs/api-reference/chat).
+
+### Verification Interfaces
+
+To ensure the integrity and trustworthiness of services, different verification mechanisms are employed. Each mechanism comes with its own specific set of protocols and requirements to ensure service verification and security.
+
+<Tabs>
+<TabItem value="tee" label="TEE" default>
+
+For TEE (Trusted Execution Environment) verification, when a service starts, it should generate a signing key within the TEE. We require CPU and GPU attestations to ensure the service is running in a Confidential VM with an NVIDIA GPU in TEE mode. These attestations should include the public key of the signing key, verifying its creation within the TEE. All inference results must be signed with this signing key.
+
+#### 1. Attestation Download Interface
+
+To facilitate attestation downloads, set up an API endpoint at:
+
+```
+GET https://{PUBLIC_IP}/attestation/report
+```
+
+This endpoint should return a JSON structure in the following format:
+
+```json
+{
+  "signing_address": "...",
+  "nvidia_payload": "..."
+}
+```
+
+_Note_: Ensure that the "nvidia_payload" can be verified using NVIDIA's [GPU Attestation API](https://docs.attestation.nvidia.com/api-docs/nras.html#post-/v3/attest/gpu).
+
```

Comment:
What is the status of decentralized TEE attestation? We may want to add another "coming soon" if we actually support this.

Reply:
Yes, added. We will support it.